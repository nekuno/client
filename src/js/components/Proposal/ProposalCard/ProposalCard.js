import PropTypes from 'prop-types';
import React, { Component } from 'react';
import RoundedImage from '../../ui/RoundedImage/RoundedImage.js';
import RoundedIcon from '../../ui/RoundedIcon/RoundedIcon.js';
import ProgressBar from '../../ui/ProgressBar/ProgressBar.js';
import styles from './ProposalCard.scss';
import translate from '../../../i18n/Translate';

@translate('ProposalCard')
export default class ProposalCard extends Component {

    static propTypes = {
        proposal      : PropTypes.object.isRequired,
        user          : PropTypes.object.isRequired,
        onClickHandler: PropTypes.func
    };

    handleClick() {
        if (this.props.onClickHandler) {
            this.props.onClickHandler();
        }
    }

    getVisualByType(type) {
        let icon = null;
        let background = null;
        switch (type) {
            case 'work':
                icon = 'paperclip';
                background = '#63CAFF';
                break;
            case 'leisure-plan':
                icon = 'send';
                background = '#D380D3';
                break;
            case 'experience-plan':
                icon = 'compass';
                background = '#7BD47E';
                break;
        }

        return {icon, background};
    }

    render() {
        const {proposal, user, strings} = this.props;
        const {type, fields} = proposal;
        const {title, description, image} = fields;
        const {username, location, age, photo, matching, similarity} = user;

        const {icon, background} = this.getVisualByType(type);

        return (
            <div className={styles.proposalCard} onClick={this.handleClick.bind(this)}>
                <div className={styles.frame}>
                    {icon ?
                        <div className={styles.type}>
                            <RoundedIcon icon={icon} size={'medium'} background={background} fontSize={'24px'}/>
                        </div>
                        : null
                    }
                    <div className={styles.proposalImage}>
                        <img src={image}/>
                        <div className={styles.topData}>
                            <h2>{title}</h2>
                        </div>
                    </div>
                    <div className={styles.userData}>
                        <RoundedImage size={'small'} url={photo.url}/>
                        <div className={styles.userText}>
                            <div className={styles.nickname}>{username}</div>
                            <div className={styles.ageCity}>{location.locality} &bull; {age}</div>
                        </div>
                    </div>
                    <div className={styles.progressBars}>
                        <div className={styles.progressBarTitle}>{strings.compatible}&nbsp;</div>
                        <div className={styles.progressBar}>
                            <ProgressBar percentage={matching} size={'small'} strokeColor={'#756EE5'}/>
                        </div>
                        <div className={styles.progressBarTitle}>{strings.similar}&nbsp;</div>
                        <div className={styles.progressBar}>
                            <ProgressBar percentage={similarity} size={'small'} strokeColor={'#756EE5'}/>
                        </div>
                    </div>
                    <div className={styles.description}>
                        <div className={styles.resumeTitle}>{strings.project}</div>
                        <div className={styles.resumeText}>{description}</div>
                    </div>
                </div>
            </div>
        );
    }
}

ProposalCard.defaultProps = {
    strings: {
        compatible: 'Compatible',
        similar   : 'Similar',
        project   : 'Project',
    }
};