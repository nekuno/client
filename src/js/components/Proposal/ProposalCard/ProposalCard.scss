@import '../../../../scss/_partials/colors';

.proposal-card {
  cursor: pointer;
  display: inline-block;
  font-weight: 600;
  margin: auto;
  position: relative;
  text-align: left;
  width: 80%;

  .frame {
    background: #FFF;
    border-radius: 10px;
    -moz-border-radius: 10px;
    -webkit-border-radius: 10px;
    box-shadow: 0 0 5px $gray-border;
    color: #6C6F82;
    font-size: 14px;
    margin: 2.5%;
    padding: 0;
    width: 95%;

    .proposal-image {
      height: 100%;
      overflow: hidden;
      position: relative;
      width: 100%;

      img {
        border-radius: 10px 10px 0 0;
        -moz-border-radius: 10px 10px 0 0;
        -webkit-border-radius: 10px 10px 0 0;
        width: 100%;
      }
      .top-data {
        bottom: 0;
        color: white;
        height: 100%;
        padding: 0 5%;
        position: absolute;
        width: 80%;

        h2 {
          bottom: 0;
          font-weight: 600;
          max-height: 80px;
          overflow: hidden;
          position: absolute;
        }
      }

    }
    .type {
      position: absolute;
      right: 16px;
      top: 16px;
    }
    .user-data {
      height: 60px;
      margin-top: 16px;
      padding: 0 5%;

      .user-text {
        display: inline-block;
        margin-left: 6px;
        vertical-align: super;

        .nickname {
          font-size: 20px;
        }
        .age-city {
          color: $gray-secondary-text;
          font-size: 14px;
          font-weight: normal;
          white-space: nowrap;
          overflow: hidden;
          text-overflow: ellipsis;
        }
      }
    }
    .progress-bars {
      margin-top: 16px;
      padding: 0 5%;

      .progress-bar-title {
        display: inline-block;
        font-size: 10px;
        height: 33px;
        line-height: 33px;
        width: 50%;

        @media only screen and (min-width: 320px) {
          font-size: 12px;
        }
        @media only screen and (min-width: 350px) {
          font-size: inherit;
        }
      }
      .progress-bar {
        display: inline-block;
        width: 40%;
      }
    }
    .resume {
      padding: 16px 5%;

      .resume-title {

      }
      .resume-text {
        font-weight: normal;
        margin-top: 16px;
      }
    }
  }
}