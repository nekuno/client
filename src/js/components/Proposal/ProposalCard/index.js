import PropTypes from 'prop-types';
import React, { Component } from 'react';
import RoundedImage from '../../ui/RoundedImage';
import styles from './ProposalCard.scss';
import translate from '../../../i18n/Translate';
import MatchingBars from "../../ui/MatchingBars";
import ProposalIcon from "../../ui/ProposalIcon";

@translate('ProposalCard')
class ProposalCard extends Component {

    static propTypes = {
        proposal      : PropTypes.object.isRequired,
        user          : PropTypes.object.isRequired,
        onClickHandler: PropTypes.func,
        // Injected by @translate:
        strings        : PropTypes.object,
    };

    static contextTypes = {
        router: PropTypes.object.isRequired
    };

    constructor(props) {
        super(props);

        this.handleClick = this.handleClick.bind(this);
    }

    handleClick() {
        this.props.onClickHandler();
    }

    renderProposalIcon(proposal) {
        let icon = '';

        switch (proposal.type) {
            case 'sports':
            case 'hobbies':
            case 'games':
                icon = 'icon-hobbie';
                break;
            case 'shows':
            case 'restaurants':
            case 'plans':
                icon = 'icon-experience';
                break;
            case 'work':
            default:
                icon = 'icon-project';
                break;
        }

        return (
            <ProposalIcon size={'medium-small'} icon={icon} background={'white'}/>
        );
    }

    render() {
        const {proposal, user, strings} = this.props;
        const {fields} = proposal;
        const {title, description} = fields;
        const proposalPhoto = fields.photo !== '' ? fields.photo : 'img/default-upload-image.png';
        const {username, location, age, photo, matching, similarity} = user;
        const locality = location && location.hasOwnProperty('locality') ? location.locality : '';

        return (
            <div className={styles.proposalCard}>
                <div className={styles.frame}>
                    <div className={styles.type}>
                        {this.renderProposalIcon(proposal)}
                    </div>
                    <div className={styles.proposalImage}>
                        <img src={proposalPhoto}/>
                    </div>
                    <div className={styles.topData}>
                        <h2>{title}</h2>
                    </div>
                    <div className={styles.userData}>
                        <RoundedImage size={'small'} url={photo.url}/>
                        <div className={styles.userText}>
                            <div className={styles.username}>{username}</div>
                            <div className={styles.ageCity}>{locality} &bull; {age}</div>
                        </div>
                    </div>
                    <MatchingBars matching={matching} similarity={similarity}/>
                    <div className={styles.description}>
                        <div className={styles.resumeTitle}>{strings.project}</div>
                        <div className={styles.resumeText}>{description}</div>
                    </div>
                </div>
            </div>
        );
    }
}

ProposalCard.defaultProps = {
    strings: {
        compatible: 'Compatible',
        similar   : 'Similar',
        project   : 'Project',
    },
    onClickHandler: () => {
    }
};

export default ProposalCard;
