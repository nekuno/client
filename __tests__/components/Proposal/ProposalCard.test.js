import React from 'react';
import ProposalCard from '../../../src/js/components/Proposal/ProposalCard/ProposalCard.js';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test ProposalCard component', () => {
    const mockFn = jest.fn();
    const proposalCard = shallow(<ProposalCard title={'Lorem ipsum dolor sit amet'}
                                               image={'http://via.placeholder.com/360x180'}
                                               type={'work'}
                                               photo={'http://via.placeholder.com/100x100'}
                                               nickname={'JohnDoe'}
                                               age={36}
                                               city={'New York'}
                                               matching={12}
                                               similarity={35}
                                               description={'Lorem ipsum dolor sit amet'}
                                               onClickHandler={mockFn}/>
    );
    it('should be defined', () => {
        expect(ProposalCard).toBeDefined();
    });
    it('should render correctly', () => {
        expect(proposalCard).toMatchSnapshot();
    });
});
