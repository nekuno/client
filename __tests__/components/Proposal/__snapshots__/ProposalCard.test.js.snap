// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ProposalCard component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TranslationComponent
    age={36}
    city="New York"
    description="Lorem ipsum dolor sit amet"
    image="http://via.placeholder.com/360x180"
    matching={12}
    nickname="JohnDoe"
    onClickHandler={[MockFunction]}
    photo="http://via.placeholder.com/100x100"
    similarity={35}
    strings={
      Object {
        "compatible": "Compatible",
        "project": "Project",
        "similar": "Similar",
      }
    }
    title="Lorem ipsum dolor sit amet"
    type="work"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "age": 36,
      "city": "New York",
      "description": "Lorem ipsum dolor sit amet",
      "image": "http://via.placeholder.com/360x180",
      "locale": undefined,
      "matching": 12,
      "nickname": "JohnDoe",
      "onClickHandler": [MockFunction],
      "photo": "http://via.placeholder.com/100x100",
      "similarity": 35,
      "strings": Object {
        "compatible": "Compatible",
        "project": "Project",
        "similar": "Similar",
      },
      "title": "Lorem ipsum dolor sit amet",
      "type": "work",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "age": 36,
        "city": "New York",
        "description": "Lorem ipsum dolor sit amet",
        "image": "http://via.placeholder.com/360x180",
        "locale": undefined,
        "matching": 12,
        "nickname": "JohnDoe",
        "onClickHandler": [MockFunction],
        "photo": "http://via.placeholder.com/100x100",
        "similarity": 35,
        "strings": Object {
          "compatible": "Compatible",
          "project": "Project",
          "similar": "Similar",
        },
        "title": "Lorem ipsum dolor sit amet",
        "type": "work",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
